compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -1470,6 +1473,60 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         false
     }
 
+    /// For code `rect::area(...)`,
+    /// if `rect` is a local variable and `area` is a valid assoc method for it,
+    /// we try to suggest `rect.area()`
+    pub(crate) fn suggest_assoc_method_call(&self, segs: &[PathSegment<'_>]) {
+        debug!("suggest_assoc_method_call segs: {:?}", segs);
+        if segs.len() != 2 {
+            return;
+        }
+        let (seg1, seg2) = (&segs[0], &segs[1]);
+        let Some(mut diag) =
+                self.tcx.sess.diagnostic().steal_diagnostic(seg1.ident.span, StashKey::CallAssocMethod)
+                else { return };
+
+        let map = self.infcx.tcx.hir();
+        let body_id = map.body_owned_by(seg1.hir_id.owner.def_id);
+        let body = map.body(body_id);
+        struct LetVisitor<'a> {
+            result: Option<&'a hir::Expr<'a>>,
+            ident_name: Symbol,
+        }
+
+        impl<'v> Visitor<'v> for LetVisitor<'v> {
```

Comment:
Why do we need to look for this local again if we already found it in `rustc_resolve` (`resolve_ident_in_lexical_scope`)?

Reply:
edit: Oh, because we need the type of the `self`... Well, we should still be able to look that up (see comment below).